pgmpy/inference/approximate_inference/Sampling.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import networkx as nx
+
+from pgmpy.models import BayesianModel
+from pgmpy.utils.mathext import sample_discrete
+from pgmpy.inference import Inference
+
+
+class BayesianModelSampling(Inference):
+    """
+    Class for sampling methods specific to Bayesian Models
+
+    Parameters
+    ----------
+    model: instance of BayesianModel
+        model on which inference queries will be computed
+
+
+    Public Methods
+    --------------
+    forward_sample(size)
+    """
+    def __init__(self, model):
+        if not isinstance(model, BayesianModel):
+            raise TypeError("Argument must be of type BayesianModel")
+        super(BayesianModelSampling, self).__init__(model)
+        self.topological_order = nx.topological_sort(model)
+        self.cpds = {}
+        for node in model.nodes():
+            self.cpds[node] = model.get_cpds(node)
+
+    def forward_sample(self, size=1):
+        """
+        Generates sample(s) from joint distribution of the bayesian network.
+
+        Parameters
+        ----------
+        size: int
+            size of sample to be generated
+
+        Returns
+        -------
+        sampled: list of dicts
+            the generated samples
+
+        Examples
+        --------
+        >>> from pgmpy.models.BayesianModel import BayesianModel
+        >>> from pgmpy.factors.CPD import TabularCPD
+        >>> student = BayesianModel([('diff', 'grade'), ('intel', 'grade')])
+        >>> cpd_d = TabularCPD('diff', 2, [[0.6], [0.4]])
+        >>> cpd_i = TabularCPD('intel', 2, [[0.7], [0.3]])
+        >>> cpd_g = TabularCPD('grade', 3, [[0.3, 0.05, 0.9, 0.5], [0.4, 0.25,
+        ...                0.08, 0.3], [0.3, 0.7, 0.02, 0.2]],
+        ...                ['intel', 'diff'], [2, 2])
+        >>> student.add_cpds(cpd_d, cpd_i, cpd_g)
+        >>> from pgmpy.inference.approximate_inference.Sampling import *
+        >>> inference = BayesianModelSampling(student)
+        >>> inference.forward_sample(2)
+        [{'intel': 'intel_0', 'grade': 'grade_1', 'diff': 'diff_1'},
+        {'intel':  'intel_1', 'grade': 'grade_0', 'diff': 'diff_0'}]
+        """
```

Comment:
@pratyakshs I don't think having a list of dicts for storing the samples is a good idea. It will unnecessarily be consuming too much memory. Maybe return a pandas dataframe or return a tuple of order of variables and the values of samples in the same order. @abinashpanda What do you think?